packages/template-retail-react-app/app/components/shopper-agent/index.jsx

```diff
@@ -1,25 +1,26 @@
 /*
- * Copyright (c) 2024, salesforce.com, inc.
+ * Copyright (c) 2025, Salesforce, Inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-
 import React, {useEffect} from 'react'
 import useScript from '@salesforce/retail-react-app/app/hooks/use-script'
 import {useUsid} from '@salesforce/commerce-sdk-react'
 import PropTypes from 'prop-types'
 import {useTheme} from '@salesforce/retail-react-app/app/components/shared/ui'
 import useMiaw from '@salesforce/retail-react-app/app/hooks/use-miaw'
 import useRefreshToken from '@salesforce/retail-react-app/app/hooks/use-refresh-token'
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
 
 const onClient = typeof window !== 'undefined'
 
 /**
  * Validates that a URL is from a trusted Salesforce domain.
  *
- * @param {string} url - The URL to validate
- * @returns {boolean} True if the URL is from a trusted Salesforce domain
+ * @param {string} url - The URL to validate (e.g., 'https://myorg.salesforce.com/script.js')
+ * @returns {boolean} True if the URL is from a trusted Salesforce domain, false otherwise
+ * @throws {TypeError} If the URL is invalid and cannot be parsed
  */
 const validateSalesforceDomain = (url) => {
```

Comment:
All of these validation functions should be moved to `app/utils/utils.js` or for shopper agents specific functions move to `app/utils/shopper-agent-utils.js`